scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala

```diff
@@ -184,6 +184,27 @@ class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
     assert(Dotty(q"trait T(a: Int)").syntax === "trait T(a: Int)")
   }
 
+  test("literalTypes") {
+    intercept[ParseException] {
+      dialects.Scala211("val a : 42 = 42").parse[Stat].get.show[Syntax]
+    }
+    val Scala211 = null // TODO: #389
+    import scala.meta.dialects.Dotty
+    assert(q"val a: 42 = 42".show[Syntax] === "val a: 42 = 42")
```

Comment:
What happens in these cases?

```scala
val works : 1 + (2 * 3) + 4 = 11
val four2 : 2 + 2 = 4
```

Taken from https://contributors.scala-lang.org/t/sip-nn-make-infix-type-alias-precedence-like-expression-operator-precedence/471